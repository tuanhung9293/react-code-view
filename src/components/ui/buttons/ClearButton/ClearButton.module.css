.base {
  display: inline-block;
  font-size: small;
  font-weight: bold;
  color: black;
  background-color: transparent;
  text-decoration: none;
  padding: 10px 15px;
  text-transform: uppercase;
  border: 0;
  border-radius: 0;
  text-align: center;
  cursor: pointer;
}

.disabled {
  opacity: 0.5;
  cursor: default;
}

.noOutline {
  outline: 0;
}