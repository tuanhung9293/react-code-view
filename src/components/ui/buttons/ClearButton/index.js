import React from 'react'
import classNames from 'classnames'
import Button from '../Button'
import styles from './ClearButton.module.css'

function ClearButton({disabled, noOutline, ...props}) {
  const className = classNames({
    [styles.base]: true,
    [styles.disabled]: disabled,
    [styles.noOutline]: noOutline
  })
  return <Button {...props} disabled={disabled} className={className} />
}

export default ClearButton