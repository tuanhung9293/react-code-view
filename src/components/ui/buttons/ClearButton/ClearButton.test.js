import React from 'react'
import {shallow} from 'enzyme'
import ClearButton from './'

it('renders as a button with children', () => {
  expect(shallow(
    <ClearButton>I am child</ClearButton>
  )).toMatchSnapshot()
})

it('overrides className', () => {
  expect(shallow(
    <ClearButton className="i-should-not-be-there">I am child</ClearButton>
  )).toMatchSnapshot()
})

it('renders as disabled', () => {
  expect(shallow(
    <ClearButton disabled>I am child</ClearButton>
  )).toMatchSnapshot()
})

it('renders as noOutline', () => {
  expect(shallow(
    <ClearButton noOutline>I am child</ClearButton>
  )).toMatchSnapshot()
})
