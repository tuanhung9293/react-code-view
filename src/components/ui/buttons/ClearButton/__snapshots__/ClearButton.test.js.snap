// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overrides className 1`] = `
<Button
  className="base"
>
  I am child
</Button>
`;

exports[`renders as a button with children 1`] = `
<Button
  className="base"
>
  I am child
</Button>
`;

exports[`renders as disabled 1`] = `
<Button
  className="base disabled"
  disabled={true}
>
  I am child
</Button>
`;

exports[`renders as noOutline 1`] = `
<Button
  className="base noOutline"
>
  I am child
</Button>
`;
